File: app/discover/scan_project.rb
# scan a project for endpoints and regions

require 'singleton'
require_relative 'api_fetch_end_points'
require_relative 'scan_region'

class ScanProject < Scanner
  include Singleton
  
  def initialize()
    super([
      {
        :type => "endpoint",
        :fetcher => ApiFetchEndPoints.new(),
        :children_scanner => ScanRegion.instance
      }
    ])
  end
  
end

File: app/discover/scan_project.py
# scan a project for endpoints and regions
from discover.scanner import Scanner
from discover.singleton import Singleton


class ScanProject(Scanner, metaclass=Singleton):
    def __init__(self):
        super(ScanProject, self).__init__([
            {
                "type": "availability_zone",
                "fetcher": "ApiFetchAvailabilityZones"
            },
            {
                "type": "host",
                "fetcher": "ApiFetchProjectHosts",
                "children_scanner": "ScanHost"
            }
        ])
